pkg/pipeline/service.go

```diff
@@ -363,3 +366,15 @@ func (s *Service) validatePipeline(cfg Config, id string) error {
 
 	return cerrors.Join(errs...)
 }
+
+func (s *Service) updateOldStatusMetrics(pl *Instance) {
+	status := strings.ToLower(pl.GetStatus().String())
+	measure.PipelinesGauge.WithValues(status).Dec()
+	measure.PipelineStatusGauge.DeleteLabels(pl.Config.Name, status)
+}
+
+func (s *Service) updateNewStatusMetrics(pl *Instance) {
+	status := strings.ToLower(pl.GetStatus().String())
+	measure.PipelinesGauge.WithValues(status).Inc()
+	measure.PipelineStatusGauge.WithValues(pl.Config.Name, status).Set(1)
+}
```

Comment:
These two refer to `before` and `after`?

Reply:
Exactly!